maths/path_length.py

```diff
@@ -0,0 +1,59 @@
+"""
+Title : Calculate the path length of a sequence of points (x,y)
+
+Description :
+        Calculates the path length of a sequence of points (x,y)
+        where the next point in the sequence follows an increasing order of x
+
+        https://math.libretexts.org/Bookshelves/Calculus/Map%3A_Calculus__Early_Transcendentals_(Stewart)/08%3A_Further_Applications_of_Integration/8.01%3A_Arc_Length
+
+"""
+
+
+def discrete_path_length(seq: list[tuple[float, float]]) -> float:
+    """
+    seq : a list of (x,y) pairs , with sorted in non decreasing manner of x
+
+    Example:
+    Generates a sequence of coordinates for a semicircle of radius and calculates
+    path length in this case. i.e length of the semicircle
+
+    >>> import numpy as np
+    >>> radius = 5
+    >>> xRange = np.arange(-radius,radius,0.001)
```

Comment:
I do not love the use of `xrange` because it existed in Python 2 and we did a LOT of work to get rid of it in Python 3.  Can we please come up with a more self-documenting variable name?